zkclient/zkclient.go

```diff
@@ -604,6 +604,8 @@ type ZKC struct {
 	ab *addressbook.AddressBook
 
 	ratchetMtx sync.Mutex
+	pendingIdentitiesMutex sync.Mutex
```

Comment:
Any reason to not use a single mutex for all of this?  This isn't exactly perf path and having multi mutexes just causes ordering issues.

Reply:
Well, they protect different things. Otherwise we could look into merging these mutexes into one, but that's beyond the scope of this PR:

$ grep -r "sync.Mutex" zkclient
zkclient/zkclient.go:	write           sync.Mutex    // connection write mutex
zkclient/zkclient.go:	ratchetMtx sync.Mutex
zkclient/zkclient.go:	pendingIdentitiesMutex sync.Mutex
zkclient/msg.go:	mtx := sync.Mutex{}               // msg and bulk mutex

Reply:
After reviewing this one more time I think this is the right design.